File: python/docmost/services/version_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse


class VersionApi:
    """Version API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_version(self) -> ApiResponse[Any]:
        return self._http.post("version", body=None)

File: go/docmost/services/version_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
)

// VersionAPI implements Version API.
type VersionAPI struct {
	http *httpclient.HTTPClient
}

func NewVersionAPI(http *httpclient.HTTPClient) *VersionAPI {
	return &VersionAPI{http: http}
}

func (a *VersionAPI) GetVersion(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "version", nil)
}

